core/src/test/scala/org/apache/spark/deploy/history/EventLogFileCompactorSuite.scala

```diff
@@ -119,7 +119,7 @@ class EventLogFileCompactorSuite extends SparkFunSuite {
     withTempDir { dir =>
       val fs = new Path(dir.getAbsolutePath).getFileSystem(hadoopConf)
 
-      val fileStatuses = writeEventsToRollingWriter(fs, "app", dir, sparkConf, hadoopConf,
+      val fileStatuses = writeEventsToRollingWriter(fs, "app a", dir, sparkConf, hadoopConf,
```

Comment:
I check this case as follows:

```
gh pr checkout 35979
manually revert change of EventLogFileCompactor.scala
mvn clean install -DskipTests -pl core -am
mvn test -pl core -Dtest=none -DwildcardSuites=org.apache.spark.deploy.history.EventLogFileCompactorSuite
```
Test result as follows:
```
Discovery starting.
Discovery completed in 1 second, 647 milliseconds.
Run starting. Expected test count is: 9
EventLogFileCompactorSuite:
- No event log files
- No compact file, less origin files available than max files to retain
- No compact file, more origin files available than max files to retain
- compact file exists, less origin files available than max files to retain
- compact file exists, number of origin files are same as max files to retain
- compact file exists, more origin files available than max files to retain
- events for finished job are dropped in new compact file
- Don't compact file if score is lower than threshold
- rewrite files with test filters
Run completed in 2 seconds, 481 milliseconds.
Total number of tests run: 9
Suites: completed 2, aborted 0
Tests: succeeded 9, failed 0, canceled 0, ignored 0, pending 0
All tests passed.
```

It seems that the new UT is successful without this pr. @lw33 

Can you help double check? @HyukjinKwon

Reply:
My mistake，i fixed this. Plz check it again.

Reply:
On second thought, I think in the production environment, illegal characters(such as %) may appear in `basedir`, maybe we can add a new case as follows:

```scala
test("SPARK-38664: ${Test display name}") {
    // create a base dir with illegal characters
    val dir = Utils.createTempDir("spark-%")
    try {
      // do test 
    } finally {
      Utils.deleteRecursively(dir)
    }
  }

```